retis/src/core/events/bpf.rs

```diff
@@ -26,42 +28,49 @@ pub(super) const COMMON_SECTION_TASK: u64 = 1;
 /// Timeout when polling for new events from BPF.
 const BPF_EVENTS_POLL_TIMEOUT_MS: u64 = 200;
 
-/// Macro that define Default-able fixed size sequence of bytes aimed
-/// to contain zero-terminated strings. Useful for unmarshaling array
-/// of characters bigger than 32 elements.
+/// Macro used to convert c_char into String.
+/// The macro returns error if the conversion fails.
 #[macro_export]
-macro_rules! event_byte_array {
-    ($name:ident, $size:expr) => {
-        #[repr(C)]
-        struct $name([u8; $size]);
-
-        impl Default for $name {
-            fn default() -> Self {
-                // Safety is respected as the type is well defined and
-                // controlled.
-                unsafe { std::mem::zeroed() }
-            }
-        }
+macro_rules! raw_to_string {
+    ($c_array:expr) => {{
+        use anyhow::{anyhow, Result};
+        use std::{ffi::CStr, os::raw::c_char};
+
+        let to_string = |arr: &[c_char]| -> Result<String> {
+            let _null_pos = arr
+                .iter()
+                .position(|&c| c == 0)
+                .ok_or_else(|| anyhow!("String is not NULL terminated"))?;
+
+            let cstr = unsafe { CStr::from_ptr(arr.as_ptr()) };
+            Ok(cstr.to_string_lossy().into_owned())
+        };
 
-        #[allow(dead_code)]
-        impl $name {
-            fn to_string(&self) -> Result<String> {
-                Ok(std::ffi::CStr::from_bytes_until_nul(&self.0)?
-                    .to_str()?
-                    .into())
-            }
+        to_string($c_array)
+    }};
+}
 
-            fn to_string_opt(&self) -> Result<Option<String>> {
-                let res = self.to_string()?;
+/// A macro for converting a `c_char` array into a `String`.  It
+/// returns an error in case of failure. Upon successful conversion,
+/// it checks if the resulting `String` is empty, returning `Ok(None)`
+/// if it is.
+#[macro_export]
+macro_rules! raw_to_string_opt {
```

Comment:
nit: as we have a single user, maybe we can open code this one? Feel free to ignore :)